### ISSUE TITLE
Can't allow a before disallowed integrated terminal

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#23362]
- Tests: #23128 
- VSCode Version: 1.11.0-insider
- OS Version: Windows

Steps to Reproduce:

1. customize a shell workspace settings using `"terminal.integrated.shell.windows": "C:\\WINDOWS\\Sysnative\\bash.exe"`
2. Disallow the shell

Try to allow the shell. The only way I found is to use the dev tools and delete the key.

**Comment by Tyriar** [comment]
Any suggestions how to surface this without annoying the user? (how would disallow differ from cancel?)

**Comment by dbaeumer** [comment]
The problem here is that after a disallow the dialog doesn't show up anymore. So no way to trust the workspace later without using the dev tools and editing local storage

**Comment by Tyriar** [comment]
@dbaeumer yes, the workarounds being moving the folder or changing local storage. The alternative is using user setting to drive this instead of local storage. Seems a bit messy doing that since paths generally aren't portable across machines and it's workspace-specific config in your user settings.

Since the use case of a user re-enabling this at a later time seems to be so small I don't think it's worth changing this. The usage numbers for workspace shell's were really small when I looked at them so the chance that a user would choose to disallow and later make a conscious decision to allow seems really slim. Closing as designed for now, will listen for feedback.
